homeassistant/components/advantage_air/climate.py

```diff
@@ -175,7 +163,12 @@ async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
 
     async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set the Fan Mode."""
-        await self.async_update_ac({"fan": HASS_FAN_MODES.get(fan_mode)})
+        mode = (
+            ADVANTAGE_AIR_MYFAN
+            if fan_mode == FAN_AUTO and self._ac.get(ADVANTAGE_AIR_AUTOFAN_ENABLED)
+            else fan_mode
+        )
```

Comment:
```suggestion
        if fan_mode == FAN_AUTO and self._ac.get(ADVANTAGE_AIR_AUTOFAN_ENABLED):
            mode = ADVANTAGE_AIR_MYFAN
        else:
            mode = fan_mode
```
In general we try to avoid multiline ternary operators